src/unity/toolkits/one_shot_object_detection/one_shot_object_detector.cpp

```diff
@@ -0,0 +1,171 @@
+#include <algorithm>
+#include <array>
+#include <cmath>
+#include <cstdio>
+#include <iostream>
+#include <limits>
+#include <numeric>
+#include <queue>
+#include <utility>
+#include <vector>
+#include <random>
+
+#include <logger/assertions.hpp>
+#include <logger/logger.hpp>
+#include <random/random.hpp>
+
+#include <boost/gil/gil_all.hpp>
+#include <boost/gil/extension/io/jpeg.hpp>
+#include <boost/gil/extension/numeric/sampler.hpp>
+#include <boost/gil/extension/numeric/resample.hpp>
+#include <boost/gil/utilities.hpp>
+
+#include <image/numeric_extension/perspective_projection.hpp>
+
+#include <Eigen/Core>
+#include <Eigen/Dense>
+
+using turi::coreml::MLModelWrapper;
+using namespace Eigen;
+using namespace boost::gil;
+
+namespace turi {
+namespace one_shot_object_detection {
+
+/* A ParameterSweep class to randomly generate different samples of parameters
+ * that can later be used to compute the transformation matrix necessary to 
+ * create image projections.
+ */
+class ParameterSweep {
+public:
+  ParameterSweep(int width, int height) {
+    width_ = width;
+    height_ = height;
+  }
+
+  double deg_to_rad(double angle) {
+    return angle * M_PI / 180.0;
+  }
+
+  double get_theta() {
+    return deg_to_rad(theta_);
+  }
+
+  double get_phi() {
+    return deg_to_rad(phi_);
+  }
+
+  double get_gamma() {
+    return deg_to_rad(gamma_);
+  }
+
+  int get_dz() {
+    return dz_;
+  }
+
+  double get_focal() {
+    return focal_;
+  }
+
+  void sample(long seed) {
+    /* Barebones */
+    theta_ = 0.0;
+    phi_ = 0.0;
+    gamma_ = 0.0;
+    std::normal_distribution<double> focal_distribution((double)width_, focal_stdev_);
+    focal_generator_.seed(seed+6);
+    focal_ = focal_distribution(focal_generator_);
+    dz_ = focal_;
+  }
+
+private:
+  int width_;
+  int height_;
+  double focal_stdev_ = 40.0;
+  std::default_random_engine focal_generator_;
+  double theta_;
+  double phi_;
+  double gamma_;
+  int dz_;
+  double focal_;
+
+};
+
+gl_sframe _augment_data(gl_sframe data, gl_sframe backgrounds, long seed) {
+  // TODO: Get input image from the data sframe.
+  // TODO: Use backgrounds from the background SFrame.
+  // TODO: Generalize 1024 and 676 to be the width and height of the image 
+  //       passed in.
+  ParameterSweep parameter_sampler = ParameterSweep(2*1024, 2*676);
+  // TODO: Take n as input.
+  int n = 1;
+  for (int i = 0; i < n; i++) {
+    parameter_sampler.sample(seed+i);
+
+    Matrix3f mat = get_transformation_matrix(2*1024, 2*676,
+      parameter_sampler.get_theta(),
+      parameter_sampler.get_phi(),
+      parameter_sampler.get_gamma(),
+      1024/2,
+      676/2,
+      parameter_sampler.get_dz(),
+      parameter_sampler.get_focal());
+
+    // TODO: Construct annotations that conform to the OD format, from the 
+    // four warped corner points.
+
+    rgb8_image_t starter_image, background;
+    // TODO: Don't hardcode this.
+    read_image("in-affine.jpg", starter_image, jpeg_tag());
```

Comment:
Any reason not to use `turi::image_util::load_image` (src/unity/lib/image_util.hpp) or perhaps `turi::read_image` (src/image/io.hpp)?

Reply:
Since this is also part of using user input, I thought I'd bring it in another PR and properly use Turi Image I/O utilities on the images passed into `data`. Should I bring it now instead?